packages/state-transition/src/util/validator.ts

```diff
@@ -83,11 +83,30 @@ export function getMaxEffectiveBalance(withdrawalCredentials: Uint8Array): numbe
 }
 
 export function getPendingBalanceToWithdraw(state: CachedBeaconStateElectra, validatorIndex: ValidatorIndex): number {
+  const pendingPartialWithdrawalsLength = state.pendingPartialWithdrawals.length;
+
+  if (pendingPartialWithdrawalsLength === 0) {
+    return 0;
+  }
+
   let total = 0;
-  for (const item of state.pendingPartialWithdrawals.getAllReadonly()) {
-    if (item.validatorIndex === validatorIndex) {
-      total += Number(item.amount);
+  let chunkStartIndex = 0;
+  const chunkSize = 100;
+
+  while (chunkStartIndex < pendingPartialWithdrawalsLength) {
+    const pendingPartialWIthdrawalChunk = state.pendingPartialWithdrawals.getReadonlyByRange(
```

Comment:
it's not useful to apply `getReadonlyByRange()` for this flow because we always have to loop the list through the end, hence the old `getAllReadonly()` has better performance

instead `getReadonlyByRange()` should be applied to flow where we may break in the middle and loading the whole list could become a waste, for example https://github.com/ChainSafe/lodestar/blob/2d43862bb537aea8ff4b91941ed8f2c2b6af3303/packages/state-transition/src/block/processWithdrawals.ts#L123

the `getReadonly()` there is ok but using `getReadonlyByRange()` is way better

Reply:
Thanks learned something new